### PR TITLE
Add select method

### PR DESCRIPTION
**Pull request by btjanaka**
## Description

Implement and test the `select` method. Depends on #5

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] Implement and test `select` method in `DancePipeline`.
- [x] Add new dependencies in `docs/requirements.txt` and `setup.py`.
- [x] Update `filter` to count the number of molecules that the pipeline handles.
- [x] Add a `get_fingerprint_from_mol` method for retrieving a fingerprint from a molecule.

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by btjanaka** [comment]
Hi @jmaat, can you take a look at this PR? Thanks!

File: dance/dance_pipeline.py [hunk 1]
```diff
@@ -100,9 +169,236 @@ def _generate_molecules_from_database(self) -> oechem.OEMol:
...
+        filtered_molecule_stream.close()
+        fingerprinted_molecule_stream.close()
+
+    def _add_single_fingerprint(self, mol: oechem.OEMol, fingerprint: "array-like of float"):
+        """Adds the fingerprint to the molecule in-place."""
+        mol.SetIntData(self.FINGERPRINT_LENGTH_NAME, len(fingerprint))
+        for i, val in enumerate(fingerprint):
+            mol.SetDoubleData(f"{self.FINGERPRINT_VALUE_NAME}_{i}", val)
+
+    def select(self,
```
Comment (btjanaka) [inline dance/dance_pipeline.py:229]: This is the core of the PR. To view the rendered version of the docstring, visit https://dance.readthedocs.io/en/btjanaka-selection/api/dance.DancePipeline.html#dance.DancePipeline.select

File: dance/dance_pipeline.py [hunk 2]
```diff
@@ -100,9 +169,236 @@ def _generate_molecules_from_database(self) -> oechem.OEMol:
...
+                    oechem.OEWriteMolecule(dataset_stream, mol)
+
+        # Clean up.
+        if dataset_type == "SMILES":
+            dataset_stream.close()
+
+        sorted_molstream.close()
+
+    @staticmethod
+    def _sort_molecules_by_fingerprint(input_oeb: pathlib.Path,
```
Comment (btjanaka) [inline dance/dance_pipeline.py:312]: I'm guessing this function will be pretty tough to understand. Let me know if you would like me to explain it in a meeting.

**Comment by btjanaka** [comment]
Fixed the bug we mentioned in our meeting. Should now be able to handle longer fingerprints.

**Comment by btjanaka** [comment]
Note: Tag as v0.1.0 when merged.

**Review by jmaat — APPROVED** [review]

**Pull request merged**